pkgs/development/python-modules/torchmetrics/default.nix

```diff
@@ -40,46 +41,47 @@ buildPythonPackage {
   ];
 
   # Let the user bring their own instance
-  buildInputs = [
-    torch
-  ];
-
-  nativeCheckInputs = [
-    pytorch-lightning
-    scikit-learn
-    scikit-image
-    cloudpickle
-    psutil
-    pytestCheckHook
-  ];
+  buildInputs = [ torch ];
 
   # A cyclic dependency in: integrations/test_lightning.py
   doCheck = false;
-  passthru.tests.check = torchmetrics.overridePythonAttrs (_: {
+
+  passthru.tests.check = torchmetrics.overridePythonAttrs {
+    pname = "${pname}-check";
+    # This is not as clean as I would like it to be, but
+    # it will work and tests.check is not user-facing
+    catchConflicts = false;
     doCheck = true;
-  });
 
-  disabledTestPaths = [
-    # These require too many "leftpad-level" dependencies
-    "tests/text"
-    "tests/audio"
-    "tests/image"
+    nativeCheckInputs = [
+      pytorch-lightning
+      scikit-learn
+      scikit-image
+      cloudpickle
+      psutil
+      pytestCheckHook
+      pytest-doctestplus
+      pytest-xdist
+    ];
 
-    # A few non-deterministic things like test_check_compute_groups_is_faster
-    "tests/bases/test_collections.py"
-  ];
+    disabledTestPaths = [
+      # These require too many "leftpad-level" dependencies
+      # Also too cross-dependent
+      "tests/unittests"
 
-  pythonImportsCheck = [
-    "torchmetrics"
-  ];
+      # A trillion import path mismatch errors
+      # Surprisingly, this isn't because of catchConflicts being off...
+      "src/torchmetrics"
+    ];
```

Comment:
Are they moved to passthru.tests on purpose? Asking because people seem to mostly keep `nativeCheckInputs`&c in the parent expression even when `doCheck = false`

Reply:
Yes.
There's no point in having ``nativeCheckInputs`` in the parent expression in this particular case.
Normally, I'd keep them though.

Reply:
Ah right, I remembered the actual reason.
IIRC it was because the inputs in nativeCheckInputs still led to the dependency hell between torchmetrics and pytorch-lightning.

Reply:
Oh it's probably about python forcing ignoring `doCheck` and using `doInstallCheck` instead (if I got the names right)

Reply:
Ah. It'll take me a while to compile to even test that but I'll let you know if changing doInstallCheck instead works.

Reply:
I found a workaround even for ``catchConflicts``.
What I'm thinking may be the case is the tests aren't really compatible with python-3.11 at this point.
We'll see after another torch build...

Reply:
Not sure why I moved it in there before, apparently it was fine?
At any rate, we don't need to turn off ``catchConflicts``, we can simply just not install the testing sub derivation.